From 8f05578d90df49dce6e13ee850fdc8bab84916ba Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Fri, 6 Oct 2017 12:23:32 +0200
Subject: [PATCH] dasdfmt: Implement '-f' for backwards compability

YaST is calling dasdfmt with '-f device', which used to be the old
calling convention. So to not keel over when used with an older
version of YaST we should accept this option, too.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 dasdfmt/dasdfmt.8 |    6 +++++-
 dasdfmt/dasdfmt.c |   10 ++++++++++
 2 files changed, 15 insertions(+), 1 deletion(-)

--- a/dasdfmt/dasdfmt.8
+++ b/dasdfmt/dasdfmt.8
@@ -11,7 +11,7 @@
 .br
         [-r \fIcylinder\fR] [-b \fIblksize\fR] [-l \fIvolser\fR] [-d \fIlayout\fR]
 .br
-        [-L] [-V] [-F] [-k] [-C] [-M \fImode\fR] \fIdevice\fR [\fIdevice\fR]
+        [-L] [-V] [-F] [-k] [-C] [-M \fImode\fR] [-f \fIdevice\fR] [\fIdevice\fR]
 
 .SH DESCRIPTION
 \fBdasdfmt\fR formats one or several DASD (ECKD) disk drive to prepare it
@@ -42,6 +42,10 @@
 Increases verbosity.
 
 .TP
+\fB-f\fR \fIdevice\fR or \fB--device\fR=\fIdevice\fR
+Specify device to format. For backwards compability only.
+
+.TP
 \fB-y\fR 
 Start formatting without further user-confirmation.
 
--- a/dasdfmt/dasdfmt.c
+++ b/dasdfmt/dasdfmt.c
@@ -113,6 +113,10 @@
 		.desc = "Format devices in parallel",
 		.flags = UTIL_OPT_FLAG_NOLONG,
 	},
+	{
+		.option = { "device", required_argument, NULL, 'f' },
+		.desc = "Specify device to format",
+	},
 	UTIL_OPT_SECTION("FORMAT OPTIONS"),
 	{
 		.option = { "blocksize", required_argument, NULL, 'b' },
@@ -1662,6 +1666,12 @@
 			}
 			g.layout_specified = 1;
 			break;
+		case 'f':
+			get_device_name(optind-1, argc, argv);
+			strncpy(g.dev_path_array[numdev], g.dev_path, strlen(g.dev_path));
+			strncpy(g.dev_node_array[numdev], g.dev_node, strlen(g.dev_node));
+			numdev++;
+			break;
 		case 'y':
 			g.withoutprompt = 1;
 			break;
